### PR TITLE
Metainfo data structure and parser

### PR DESCRIPTION
**Pull request by synlestidae**
Feel free to leave feedback and edit as needed

### CODE REVIEW

File: src/metainfo.rs [hunk 1]
```diff
@@ -0,0 +1,222 @@
+use bencode::{Bencode, BDict, BList, BString, BInt};
+use std::{error, fmt};
+
+#[derive(Default)]
+pub struct MetaInfo {
+    pub announce: String,
+    pub announce_list: Vec<String>,
```
Comment (MatejLach) [inline src/metainfo.rs]: I think this is supposed to be a list of lists of strings, as in: `Vec<Vec<String>>` as per https://wiki.theory.org/BitTorrentSpecification#Metainfo_File_Structure 
I also looked at this [Arch Linux torrent](https://www.archlinux.org/releng/releases/2016.08.01/torrent/) inside [QT Torrent Editor](https://sourceforge.net/projects/torrent-file-editor) ('Raw' tab) and I see nested lists:

    "announce-list": [
        [
            "http://tracker.archlinux.org:6969/announce"
        ]
    ],

**Comment by MatejLach** [comment]
@synlestidae Great work, as always 👍 ,however I'll investigate if #8 can help here, before merging.

**Comment by synlestidae** [comment]
Yeah. Addressing #8 would let me get rid of `fn get_u32_or_error` and related functions.

**Comment by synlestidae** [comment]
All right I finally got around to it. Got it parsing two metainfo files. It uses TryFrom (of which I rolled my own). There is a file in this pull request, src/tests/.mod.rs.swp, that I can't delete from it since git can't seem to find it >:(

**Comment by MatejLach** [comment]
This is amazing work as always, thanks a lot!

**Pull request merged**